### PR TITLE
KYLIN-4294: Add http api for metrics

### PR DESCRIPTION
**Pull request by zhangxiang17**
1.  Expose metrics through http api to facilitate the integration of some external monitoring components, such as tsdb
2. add a python script for tcollector

### CODE REVIEW

**Comment by hit-lacus** [comment]
Good idea, please give me some time, and I will do a test for it.

**Comment by hit-lacus** [comment]
I have a suggestion, please sumbit a doc to introduce how to monitor Kylin by TimeSeries DB in your way.  Better with some screenshot!

**Comment by zhangxiang17** [comment]
I will paste some screenshot later, @hit-lacus thanks very much ^_^

**Comment by zhangayqian** [comment]
@zhangxiang17 Hi, I have tested for this commit. 
The API works well. 
But failed when I use the python file to get the metrics because `'no-cache, no-store, max-age=0, must-revalidate'`. 
Maybe the authorization should be added to Request？Or am I using it in the wrong way?

**Comment by zhangxiang17** [comment]
> @zhangxiang17 Hi, I have tested for this commit.
> The API works well.
> But failed when I use the python file to get the metrics because `'no-cache, no-store, max-age=0, must-revalidate'`.
> Maybe the authorization should be added to Request？Or am I using it in the wrong way?

@zhangayqian sorry, it is a my fault to forget add permission in `testing` profile, and I only added it in `saml` profile, so i will add it later, sorry:(

**Comment by zhangxiang17** [comment]
image

this is our monitor page based on grafana and tsdb.

File: server/src/main/resources/kylinSecurity.xml [hunk 1]
```diff
@@ -241,6 +241,7 @@
             <scr:intercept-url pattern="/api/query*/**" access="isAuthenticated()"/>
             <scr:intercept-url pattern="/api/metadata*/**" access="isAuthenticated()"/>
             <scr:intercept-url pattern="/api/**/metrics" access="permitAll"/>
+            <scr:intercept-url pattern="/api/**/jmetrics/**" access="permitAll"/>
```
Comment (nichunen) [inline server/src/main/resources/kylinSecurity.xml]: Why make this api permitAll？

**Pull request merged**